cmd/server/main.go

```diff
@@ -52,49 +53,60 @@ func buildCLI() *cli.App {
 	app.Name = "temporal"
 	app.Usage = "Temporal server"
 	app.Version = headers.ServerVersion
-
+	app.ArgsUsage = " "
 	app.Flags = []cli.Flag{
-		cli.StringFlag{
-			Name:   "root, r",
-			Value:  ".",
-			Usage:  "root directory of execution environment",
-			EnvVar: config.EnvKeyRoot,
+		&cli.StringFlag{
+			Name:    "root",
+			Aliases: []string{"r"},
+			Value:   ".",
+			Usage:   "root directory of execution environment",
+			EnvVars: []string{config.EnvKeyRoot},
 		},
-		cli.StringFlag{
-			Name:   "config, c",
-			Value:  "config",
-			Usage:  "config dir path relative to root",
-			EnvVar: config.EnvKeyConfigDir,
+		&cli.StringFlag{
+			Name:    "config",
+			Aliases: []string{"c"},
+			Value:   "config",
+			Usage:   "config dir path relative to root",
+			EnvVars: []string{config.EnvKeyConfigDir},
 		},
-		cli.StringFlag{
-			Name:   "env, e",
-			Value:  "development",
-			Usage:  "runtime environment",
-			EnvVar: config.EnvKeyEnvironment,
+		&cli.StringFlag{
+			Name:    "env",
+			Aliases: []string{"e"},
+			Value:   "development",
+			Usage:   "runtime environment",
+			EnvVars: []string{config.EnvKeyEnvironment},
 		},
-		cli.StringFlag{
-			Name:   "zone, az",
-			Usage:  "availability zone",
-			EnvVar: config.EnvKeyAvailabilityZone,
+		&cli.StringFlag{
+			Name:    "zone",
+			Aliases: []string{"az"},
+			Usage:   "availability zone",
+			EnvVars: []string{config.EnvKeyAvailabilityZone},
 		},
 	}
 
-	allServicesStringSlice := cli.StringSlice(temporal.Services)
-	app.Commands = []cli.Command{
+	app.Commands = []*cli.Command{
 		{
-			Name:  "start",
-			Usage: "Start Temporal server",
+			Name:      "start",
+			Usage:     "Start Temporal server",
+			ArgsUsage: " ",
 			Flags: []cli.Flag{
-				cli.StringSliceFlag{
-					Name:  "services, s",
-					Value: &allServicesStringSlice,
-					Usage: "List of services to start",
+				&cli.StringSliceFlag{
+					Name:    "services",
+					Aliases: []string{"s"},
+					Value:   cli.NewStringSlice(temporal.Services...),
+					Usage:   "List of services to start",
 				},
 			},
+			Before: func(c *cli.Context) error {
+				if c.Args().Len() > 0 {
+					return cli.NewExitError("ERROR: start command doesn't support arguments. Use --services flags instead.", 1)
+				}
+				return nil
```

Comment:
This will generate an error and exit in case of any wrong arguments/flags are passed. Instead of blindly running all services as it was before.